addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java

```diff
@@ -113,22 +150,26 @@ public void testExtractTextOperation() throws Exception {
              .set("removepatternfromresult", false);
         DocumentModel documentModified = (DocumentModel) automationService.run(ctx, chain);
         assertEquals("Contract Number: 123456789", documentModified.getPropertyValue("dc:description"));
+        
+        // Getting part of a line
         chain = new OperationChain("testChain");
         chain.add(PDFExtractTextOperation.ID)
              .set("save", true)
              .set("targetxpath", "dc:description")
-             .set("patterntofind", "Something that is not in the file")
-             .set("removepatternfromresult", false);
+             .set("patterntofind", "Contract Number: ")
+             .set("removepatternfromresult", true);
         documentModified = (DocumentModel) automationService.run(ctx, chain);
-        assertNull(documentModified.getPropertyValue("dc:description"));
+        assertEquals("123456789", documentModified.getPropertyValue("dc:description"));
+        
+        // Passing a pattern that does not exist => extracted text must be null
         chain = new OperationChain("testChain");
         chain.add(PDFExtractTextOperation.ID)
              .set("save", true)
              .set("targetxpath", "dc:description")
-             .set("patterntofind", "Contract Number: ")
-             .set("removepatternfromresult", true);
+             .set("patterntofind", "Something that is not in the file")
+             .set("removepatternfromresult", false);
         documentModified = (DocumentModel) automationService.run(ctx, chain);
-        assertEquals("123456789", documentModified.getPropertyValue("dc:description"));
+        assertNull(documentModified.getPropertyValue("dc:description"));
```

Comment:
You could also factorize the 3 cases, though that's just improving some existing code.

Reply:
Ok, done in next commit(s)